### PR TITLE
Use Data.Tree instead of Rose.

### PR DESCRIPTION
**Pull request by stevana**

### CODE REVIEW

File: example/test/MutableReferenceSpec.hs [hunk 1]
```diff
@@ -115,13 +116,13 @@ prop_shrinkForkMinimal = shrinkPropertyHelper (prop_parallel RaceCondition) $ \o
     = anyRose isMinimal
     . rose (liftShrinkFork shrink1)
     where
-    anyRose :: (a -> Bool) -> Rose a -> Bool
-    anyRose p (Rose x xs) = p x || any (anyRose p) xs
+    anyRose :: (a -> Bool) -> Tree a -> Bool
+    anyRose p (Node x xs) = p x || any (anyRose p) xs
```
Comment (Danten) [inline example/test/MutableReferenceSpec.hs]: `anyRose p = foldTree (\x y -> p x || or y)`

File: example/test/MutableReferenceSpec.hs [hunk 2]
```diff
@@ -115,13 +116,13 @@ prop_shrinkForkMinimal = shrinkPropertyHelper (prop_parallel RaceCondition) $ \o
...
 
-    rose :: (a -> [a]) -> a -> Rose a
+    rose :: (a -> [a]) -> a -> Tree a
```
Comment (Danten) [inline example/test/MutableReferenceSpec.hs]: `rose f = unfoldTree (id &&& f)`

Comment (stevana) [inline example/test/MutableReferenceSpec.hs]: Fixed, thanks!

File: example/test/MutableReferenceSpec.hs (hunk 1 above)
Comment (stevana) [inline example/test/MutableReferenceSpec.hs]: `foldTree` isn't part of `Data.Tree` in the `containers` version we are using... 

(Also an unfold followed by a fold is suggest we really need `hyloTree` :-).

**Pull request merged**